app/models.py

```diff
@@ -0,0 +1,197 @@
+from datetime import datetime
+
+from flask import current_app, url_for
+from flask_login import UserMixin
+from itsdangerous import TimedJSONWebSignatureSerializer as Serializer
+from werkzeug.security import check_password_hash, generate_password_hash
+
+from . import db
+
+# This is a db table that keeps the many to many relationship between
+# Short URLs and their Visitors
+visits = db.Table('visits',
+                  db.Column('visitor_id', db.Integer,
+                            db.ForeignKey('visitors.visitor_id')),
+                  db.Column('short_url_id', db.Integer,
+                            db.ForeignKey('short_urls.short_url_id'))
+                  )
+
+
+class User(UserMixin, db.Model):
+    """
+    This is the User model class which contains a user information an has a
+    one to many relationship with the ShortUrl model.
+    """
+    __tablename__ = 'users'
+    user_id = db.Column(db.Integer, primary_key=True)
+    first_name = db.Column(db.String(30), index=True)
+    last_name = db.Column(db.String(30), index=True)
+    email = db.Column(db.String(64), unique=True)
+    password_hash = db.Column(db.String(128))
+    short_urls = db.relationship('ShortUrl', backref='user')
+    anonymous = False
+
+    def get_id(self):
+        # This returns the id of this user
+        return self.user_id
+
+    @property
+    def is_anonymous(self):
+        # This returns the anonymity status of the user
+        return self.anonymous
+
+    @property
+    def password(self):
+        # This function raise an error for accessing the user password
+        raise AttributeError('password is not a readable attribute')
+
+    def generate_auth_token(self, expiration):
+        # This generates a user token base on the user id
+        s = Serializer(
+            current_app.config['SECRET_KEY'], expires_in=expiration)
+        return s.dumps({'user_id': self.user_id})
+
+    @staticmethod
+    def verify_auth_token(token):
+        # This function verifies the token supplied by the user
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except:
+            return None
```

Comment:
I actually didn't want them to know because I am thinking telling someone who wants to hack the system such information will be like pointing them into the right direction.